wallet-wasm/src/lib.rs

```diff
@@ -1254,3 +1255,71 @@ pub extern "C" fn decrypt_with_password(
         -1
     }
 }
+
+#[no_mangle]
+pub extern "C" fn redemption_private_to_address(
+    private_ptr: *const c_uchar,
+    protocol_magic: u32,
+    out: *mut c_uchar,
+) -> u32 {
+    let priv_key = unsafe {
+        let slice: &[u8] = std::slice::from_raw_parts(private_ptr, redeem::PRIVATEKEY_SIZE);
+        redeem::PrivateKey::from_slice(slice).unwrap()
+    };
+    let pub_key = priv_key.public();
+    let addr_type = address::AddrType::ATRedeem;
```

Comment:
What about redeem_pubkey_to_txid() in cardano's tx.rs?

Reply:
Oh this is awesome 👍 Simplified the function